modules/raft/src/integrationTest/java/org/apache/ignite/internal/raftsnapshot/ItParallelRaftSnapshotsTest.java

```diff
@@ -0,0 +1,138 @@
+                TEST_ZONE_NAME, DEFAULT_STORAGE_PROFILE
+        );
+
+        executeSql(zoneSql);
+
+        String tableSql = String.format(
+                "CREATE TABLE %s (key INT PRIMARY KEY, val VARCHAR(20)) ZONE %s ",
+                TEST_TABLE_NAME, TEST_ZONE_NAME
+        );
+
+        executeSql(tableSql);
+
+        ReplicationGroupId groupId = cluster.solePartitionId(TEST_ZONE_NAME, TEST_TABLE_NAME);
+
+        int primaryReplicaIndex = primaryReplicaIndex(groupId);
+
+        // Stop two nodes that are neither primary replicas for the test table nor for the metastorage.
+        List<Integer> nodesToKill = cluster.aliveNodesWithIndices().stream()
+                .map(IgniteBiTuple::get1)
+                .filter(index -> index != primaryReplicaIndex && index != 0)
+                .limit(2)
+                .collect(toList());
+
+        nodesToKill.parallelStream().forEach(cluster::stopNode);
+
+        // After the nodes have been stopped, insert some data and truncate the Raft log on primary replica.
+        executeSql(primaryReplicaIndex, String.format("INSERT INTO %s VALUES (1, 'one')", TEST_TABLE_NAME));
+
+        validateTableData(primaryReplicaIndex);
+
+        truncateLog(primaryReplicaIndex, groupId);
+
+        // Start the nodes in parallel, Raft leader is expected to start streaming snapshots on them.
+        nodesToKill.parallelStream().forEach(cluster::startNode);
+
+        nodesToKill.forEach(this::validateTableData);
+    }
+
+    private int primaryReplicaIndex(ReplicationGroupId groupId) {
+        IgniteImpl node = unwrapIgniteImpl(cluster.aliveNode());
+
+        CompletableFuture<ReplicaMeta> primaryFuture = node.placementDriver()
+                .awaitPrimaryReplica(groupId, node.clockService().now(), 30, SECONDS);
+
+        assertThat(primaryFuture, willBe(notNullValue()));
+
+        String primaryName = primaryFuture.join().getLeaseholder();
+
+        assertThat(primaryName, is(notNullValue()));
+
+        return cluster.nodeIndex(primaryName);
+    }
+
+    private void truncateLog(int nodeIndex, ReplicationGroupId groupId) {
+        IgniteImpl node = unwrapIgniteImpl(cluster.node(nodeIndex));
+
+        Member member = Member.votingMember(node.name());
+
+        CompletableFuture<Void> truncateFuture = node.replicaManager().replica(groupId)
+                .thenCompose(replica -> replica.createSnapshotOn(member, true));
+
+        assertThat(truncateFuture, willCompleteSuccessfully());
+    }
+
+    private void validateTableData(int nodeIndex) {
+        Table table = cluster.node(nodeIndex).tables().table(TEST_TABLE_NAME);
+
+        InternalTable internalTable = unwrapTableImpl(table).internalTable();
+
+        MvPartitionStorage mvPartition = internalTable.storage().getMvPartition(0);
+
+        assertThat(mvPartition, is(notNullValue()));
+
+        await().until(mvPartition::estimatedSize, is(1L));
```

Comment:
Waiting without timeout?

Reply:
It's not without a timeout, it uses the default awaitility's timeout, which is 10 seconds